### PR TITLE
Fix locator moving on screen touch when mouse not treated as touch screen

### PR DESCRIPTION
**Pull request by nirvn**
@3nids , this fixes one of the two issues I sent you videos of on WhatsApp.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-220f4893eeffe15fd731a894fafffdaa505065d8-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-220f4893eeffe15fd731a894fafffdaa505065d8-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-220f4893eeffe15fd731a894fafffdaa505065d8-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-220f4893eeffe15fd731a894fafffdaa505065d8-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c3290a21d272d458d57f83719a43506751b49035-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c3290a21d272d458d57f83719a43506751b49035-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c3290a21d272d458d57f83719a43506751b49035-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c3290a21d272d458d57f83719a43506751b49035-x86_64.apk)

File: src/qml/MapCanvas.qml [hunk 1]
```diff
@@ -106,7 +106,7 @@ Item {
     // stylus clicked
     TapHandler {
       enabled: !mouseAsTouchScreen
-      acceptedDevices: PointerDevice.AllDevices & ~PointerDevice.TouchScreen
+      acceptedDevices: PointerDevice.Stylus
```
Comment (3nids) [inline src/qml/MapCanvas.qml:109]:
```suggestion
      acceptedDevices: PointerDevice.Stylus | DeviceType.Mouse | DeviceType.TouchPad
```
The thing is that I would like to be able to test the stylus behavior with the mouse/trackpad for debugging

Comment (nirvn) [reply src/qml/MapCanvas.qml:109]: @3nids , I've tried stylus + mouse + touchpad 2 commits ago, didn't change anything. I'm removing one item at a time to see if we can go anywhere :)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c35b1ccf5290812c28d5e93ce7ff1cadf1984323-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c35b1ccf5290812c28d5e93ce7ff1cadf1984323-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c35b1ccf5290812c28d5e93ce7ff1cadf1984323-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-c35b1ccf5290812c28d5e93ce7ff1cadf1984323-x86_64.apk)

**Comment by nirvn** [comment]
@3nids , OK, I've eliminated all pointerdevice one by one, until stylus was left, to no good.

I'm trying to remove the element altogether, let's see. Focusing on this at this very moment as I've noticed the issue is only there if the mouse isn't treated as touch. If you switch that setting on on your android device, the issue goes away.

**Comment by 3nids** [comment]
>  If you switch that setting on on your android device, the issue goes away.

that sounds correct, since the "stylus handler" is enabled only when the setting `mouseAsTouchScreen ` is false.

What is your conclusion to this? A bug on Qt side?

**Comment by nirvn** [comment]
@3nids , if the regression is gone by removing the element altogether, I think we'd be left with only a few options, the only one that sounds good to me is to initiate the setting to true or false based on device running QField.

I.e., if we detect Android, make that setting true by default. If we detect windows/linux, make that setting false by default. Maybe revisiting if things work better when we're on 5.14.

Also, if it is a bug with TapHandler, god, it's a poorly implemented element (considering I ran into my own set of issues with legend just a week ago).

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-8c889cc01508a3fffdaf2712ad84bac6a8f20376-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-8c889cc01508a3fffdaf2712ad84bac6a8f20376-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-8c889cc01508a3fffdaf2712ad84bac6a8f20376-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-8c889cc01508a3fffdaf2712ad84bac6a8f20376-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-cbbe61de42024ea02a3ec6b3a6df6e10752ed8d5-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-cbbe61de42024ea02a3ec6b3a6df6e10752ed8d5-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-cbbe61de42024ea02a3ec6b3a6df6e10752ed8d5-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-cbbe61de42024ea02a3ec6b3a6df6e10752ed8d5-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-25b21a5a6fb179ede0df5b5dbe2b37b942d854b4-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-25b21a5a6fb179ede0df5b5dbe2b37b942d854b4-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-25b21a5a6fb179ede0df5b5dbe2b37b942d854b4-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-25b21a5a6fb179ede0df5b5dbe2b37b942d854b4-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-53eeb65cc119c0827ef3d93dadbe2bcedc748048-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-53eeb65cc119c0827ef3d93dadbe2bcedc748048-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-53eeb65cc119c0827ef3d93dadbe2bcedc748048-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-53eeb65cc119c0827ef3d93dadbe2bcedc748048-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-88217a596c3bb91f7bccc784cc493d6a1d052631-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-88217a596c3bb91f7bccc784cc493d6a1d052631-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-88217a596c3bb91f7bccc784cc493d6a1d052631-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-88217a596c3bb91f7bccc784cc493d6a1d052631-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-03543304c27ee3c706fb1ff9d826559f24dec748-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-03543304c27ee3c706fb1ff9d826559f24dec748-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-03543304c27ee3c706fb1ff9d826559f24dec748-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-03543304c27ee3c706fb1ff9d826559f24dec748-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-266d92a3a2573dd61716b7e3e3b2dc4150d2d2e6-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-266d92a3a2573dd61716b7e3e3b2dc4150d2d2e6-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-266d92a3a2573dd61716b7e3e3b2dc4150d2d2e6-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-266d92a3a2573dd61716b7e3e3b2dc4150d2d2e6-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-5b269cd2857233c21f59133c4bba88502d2a3fc1-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-5b269cd2857233c21f59133c4bba88502d2a3fc1-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-5b269cd2857233c21f59133c4bba88502d2a3fc1-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-5b269cd2857233c21f59133c4bba88502d2a3fc1-x86_64.apk)

**Comment by nirvn** [comment]
@3nids , OK, finally fixed the regression.

Big TIL: by default, Qt will synthesize mouse events for unhandled touche events. The behavior can be toggled on / off via the AA_SynthesizeMouseForUnhandledTouchEvents application app. 

For *Handler elements, it means that if we want to exclude a touch but not a mouse, we need to add another handler specifically for touch to avoid that being transformed into a mouse (making the ~PointerDevice.TouchScreen pointless).

Ouf.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3307ef5ec764d5de01e472406c15c90160ec9c58-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3307ef5ec764d5de01e472406c15c90160ec9c58-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3307ef5ec764d5de01e472406c15c90160ec9c58-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3307ef5ec764d5de01e472406c15c90160ec9c58-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ed766472f795857e78ac3086f4e5de9fcfb2e82-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ed766472f795857e78ac3086f4e5de9fcfb2e82-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ed766472f795857e78ac3086f4e5de9fcfb2e82-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ed766472f795857e78ac3086f4e5de9fcfb2e82-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-f61fa825289e691f97281f456b71bc3f09cf22f4-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-f61fa825289e691f97281f456b71bc3f09cf22f4-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-f61fa825289e691f97281f456b71bc3f09cf22f4-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-f61fa825289e691f97281f456b71bc3f09cf22f4-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-012718bcefd89f14de11d5204b6d6c11b2766413-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-012718bcefd89f14de11d5204b6d6c11b2766413-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-012718bcefd89f14de11d5204b6d6c11b2766413-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-012718bcefd89f14de11d5204b6d6c11b2766413-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-bf9738c47b53cd27d04ec34d37155912ec41e140-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-bf9738c47b53cd27d04ec34d37155912ec41e140-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-bf9738c47b53cd27d04ec34d37155912ec41e140-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-bf9738c47b53cd27d04ec34d37155912ec41e140-x86_64.apk)

**Comment by 3nids** [comment]
is it working???

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-0507a7fbeaad3c812598560903e5af2ad5cdfe04-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-0507a7fbeaad3c812598560903e5af2ad5cdfe04-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-0507a7fbeaad3c812598560903e5af2ad5cdfe04-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-0507a7fbeaad3c812598560903e5af2ad5cdfe04-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-65fd5758dd7099e19032c0ce6011d1d46571cc3a-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-65fd5758dd7099e19032c0ce6011d1d46571cc3a-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-65fd5758dd7099e19032c0ce6011d1d46571cc3a-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-65fd5758dd7099e19032c0ce6011d1d46571cc3a-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ee99eea136c925d4f00aa33069389dda272b2c8-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ee99eea136c925d4f00aa33069389dda272b2c8-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ee99eea136c925d4f00aa33069389dda272b2c8-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-3ee99eea136c925d4f00aa33069389dda272b2c8-x86_64.apk)

**Comment by nirvn** [comment]
OK, nightmare over, regression fixed. This can be merged.

File: src/qml/qgismobileapp.qml [hunk 2]
```diff
@@ -159,14 +161,34 @@ ApplicationWindow {
...
+
+        }
+
+        onHoveredChanged: {
+            if ( !hovered )
+              coordinateLocator.sourceLocation = undefined
+        }
+    }
+
+    HoverHandler {
```
Comment (3nids) [inline src/qml/qgismobileapp.qml:175]: it might be worth adding a comment on the logic followed

File: src/qml/qgismobileapp.qml [hunk 3]
```diff
@@ -172,6 +172,10 @@ ApplicationWindow {
         }
     }
 
+    /* The second hover handler is a workaround what appears to be an issue with
+     * Qt whereas synthesized mouse event would trigger the first HoverHandler EventPoint
+     * through Pointerevice.TouchScreen was explicitly taken out of the accepted devices.
```
Comment (3nids) [inline src/qml/qgismobileapp.qml:177]:
```suggestion
     * though Pointerevice.TouchScreen was explicitly taken out of the accepted devices.
```
?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-275066f9d764f65509aa02434d84df357a427872-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-275066f9d764f65509aa02434d84df357a427872-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-275066f9d764f65509aa02434d84df357a427872-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1005-275066f9d764f65509aa02434d84df357a427872-x86_64.apk)

**Comment by 3nids** [comment]
you're the man!

**Pull request merged**